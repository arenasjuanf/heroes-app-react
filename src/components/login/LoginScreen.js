import React from 'react'

export const LoginScreen = ({history}) => {


        
    return (
        <div className="">
            <h1>login</h1>
            <hr />

            <button 
                type="button" 
                className="btn btn-primary"
                onClick={ ()=> history.replace('/') }
            >
                Login
            </button>

        </div>
    )
}
