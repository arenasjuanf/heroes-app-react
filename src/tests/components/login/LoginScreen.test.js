import { mount } from "enzyme"
import { AuthContext } from "../../../auth/AuthContext";
import { LoginScreen } from "../../../components/login/LoginScreen"
import { types } from "../../../types/types";

describe("Pruebas en <LoginScreen />", () => {

    const history = {
        replace: jest.fn()
    }

    const contextValue = {
        dispatch: jest.fn(),
        replace: jest.fn(),
        user:{
            logged: false
        }
    }
    
    const wrapper = mount(
        <AuthContext.Provider value={contextValue} >
            <LoginScreen history={history}/>
        </AuthContext.Provider>
    
    );

    test("debe mostrarse correctamente", () => {
        

        expect(wrapper).toMatchSnapshot()
    })

    test("debe de realizar dispatch y navegaciÃ³n", () => {

        wrapper.find('button').prop('onClick')();
        expect(contextValue.dispatch).toHaveBeenCalledWith({
            type: types.login,
            payload:{
                name: "juan felipe arenas"
            }
        })

        expect(contextValue.replace).not.toHaveBeenCalled();

    })


})