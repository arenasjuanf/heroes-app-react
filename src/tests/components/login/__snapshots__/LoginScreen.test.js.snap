// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen /> debe mostrarse correctamente 1`] = `
<div
  className="container"
>
  <h1>
    login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
    type="button"
  >
    Login
  </button>
</div>
`;
